src/cocaine-app/yt_worker.py

```diff
@@ -0,0 +1,170 @@
+                    for error in result.errors:
+                        logger.error("YQL query result error: %s", str(error))
+
+                    time.sleep(self._delay)
+                except:
+                    logger.exception("YQL request excepted %s", query)
+                    return result
+
+        logger.exception("YQL request attempts has exhausted (%s)", str(self._attempts))
+        return result
+
+    def request_expired_stat(self, aggregate_table, expired_threshold, timeout=None):
+        """
+        Send request to aggregate table to find volume of expired space in couples
+        :param aggregate_table: the name of aggregation table
+        :param expired_threshold: a couples with expired volume above param are suspected
+        :return list of couples with expired date more then trigger
+        """
+
+        timestamp = int(time.time())
+        aggregate_query = """
+        SELECT couple_id
+        FROM
+            (SELECT
+                        couple_id,
+                        SUM(expired_size) AS sum_expired_size
+                    FROM [{table}]
+                    WHERE expiration_date <= {timestamp}
+                    GROUP BY couple_id)
+            WHERE  sum_expired_size >= {trigger};
+        """.format(table=aggregate_table, timestamp=timestamp, trigger=expired_threshold)
+
+        r = self.send_request(aggregate_query, timeout)
+        if r.rows_count == 0:
+            logger.error("empty result")
+            return None
+
+        valid_couples = []
+
+        for table in r.results:  # access to results blocks until they are ready
+            table.fetch_full_data()
+            for row in table.rows:
+                try:
+                    couple = int(unicode(row[0]))
+                except ValueError:
+                    logger.exception("couple %s is invalid", couple)
+                else:
+                    logger.debug("couple %d is valid", couple)
+                    valid_couples.append(couple)
+
+        return valid_couples
+
+    def prepare_aggregate_for_yesterday(self, aggregate_table):
+        """
+        Update aggregate table from YT with data from yesterday logs
+        :param aggregate_table: the name of aggregate table
+        :return:
+        """
+        yesterday = datetime.date.today() - datetime.timedelta(days=1)
+        return self.prepare_aggregate_table(aggregate_table, yesterday)
+
+    def prepare_aggregate_table(self, aggregate_table, date):
+        """
+        Update aggregate table from YT. The processing may take up to ... time
+        :param aggregate_table: the name of aggregate_table (where to store data)
+        :param date: data from which day to use for update
+        :param expired_threshold: a couples with expired volume above param are suspected
+        """
+        date_iso = date.isoformat()
+        main_table_per_day = "statbox/mds-proxy-log/{}".format(date_iso)
```

Comment:
Fixed in https://github.com/yandex/mastermind/pull/311/commits/f7c40b6c6935d1d3f60b4dd59f2e3df4c76114eb